cmd/pinniped-proxy/src/https.rs

```diff
@@ -0,0 +1,124 @@
+use anyhow::Result;
+use hyper::{HeaderMap, header::HeaderValue};
+use url::Url;
+
+const DEFAULT_K8S_API_SERVER_URL: &str = "https://kubernetes.local";
+const HEADER_K8S_API_SERVER_URL: &str = "PINNIPED_PROXY_API_SERVER_URL";
+const INVALID_SCHEME_ERROR: &'static str = "invalid scheme, https required";
+
+/// validate_url returns a result containing the validated url or an error if it is invalid.
+fn validate_url(u: String) -> Result<String> {
+    let result = Url::parse(&u);
+    match result {
+        Ok(url) => match url.scheme() {
+            "https" => Ok(u),
+            _ => Err(anyhow::anyhow!(INVALID_SCHEME_ERROR)),
```

Comment:
What's the meaning of this underscore?

Reply:
You mentioned it in another PR:

> The _ is just to avoid warnings about unused variables (I'll use the in later PRs), similar to golang (though go forces you rather than warns).

Thanks!